### PR TITLE
Mitchell lester

### PR DESCRIPTION
**Pull request by lester2684**
update matrix unit test

### CODE REVIEW

File: jme3-core/src/main/java/com/jme3/math/Matrix.java [hunk 1]
```diff
@@ -238,11 +238,11 @@ public float get(int i, int j)
     {
         for (int c = 0; c < M; c++)
         {
-            for (int r = 0; r < M; r++)
+            for (int r = 0; r< M; r++)
             {
             	if(r == i && c == j)
             	{
-            		return matrix[r][c];
+            		return matrix[c][r];
```
Comment (aaronang) [inline jme3-core/src/main/java/com/jme3/math/Matrix.java:245]: What's the reason for this change?

File: jme3-core/src/main/java/com/jme3/math/MatrixTest.java [hunk 2]
```diff
@@ -0,0 +1,1059 @@
...
+
+import java.util.Vector;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class MatrixTest {
+
+	@Before
+	public void setUp() throws Exception {
```
Comment (aaronang) [inline jme3-core/src/main/java/com/jme3/math/MatrixTest.java]: You can remove this method if it is empty.

**Comment by aaronang** [comment]
I find it quite difficult to review, since you changed 86 files. Anyways, the test methods can benefit from better naming and the variables too. For example, `matrix4_1`, what does it mean?

File: jme3-core/src/main/java/com/jme3/math/Matrix.java [hunk 3]
```diff
@@ -2125,27 +2125,9 @@ public void scale(Vector3f scale)
...
-            		if (Math.abs(mat.matrix[c][r]) > 1e-4)
-            		{
-            			return false;
-            		}
-                }
-            }
-        }
-        return Math.abs(mat.matrix[3][2]) <= 1e-4;
+    	Matrix identity = new Matrix(mat.M);
+    	mat.equals(identity);
```
Comment (aaronang) [inline jme3-core/src/main/java/com/jme3/math/Matrix.java]: Why is this line here? What does it do?

File: jme3-core/src/main/java/com/jme3/math/Matrix.java [hunk 4]
```diff
@@ -2125,27 +2125,9 @@ public void scale(Vector3f scale)
...
+        return mat.equals(identity);
```
Comment (aaronang) [inline jme3-core/src/main/java/com/jme3/math/Matrix.java]: I think this refactor is incorrect. This method checks if a matrix is an identity matrix. What you did here: you always return `true`, which shouldn't be the case.

**Comment by aaronang** [comment]
As a suggestion, I think it is nice to provide another constructor for `Matrix`, e.g. `Matrix(int n, int m)`, such that one is able to create a 4 by 3 matrix. This way we make use of constructor overloading :wink:

File: jme3-core/src/main/java/com/jme3/math/Matrix.java (hunk 1 above)
Comment (lester2684) [inline jme3-core/src/main/java/com/jme3/math/Matrix.java:245]: It was working incorrectly.

**Comment by lester2684** [comment]
In regards to the constructor Matrix(int n, int m), I think some of the methods only work on nxn matrices
